bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/scanner/RowMerger.java

```diff
@@ -107,28 +112,54 @@ public boolean isRowCommitted() {
   }
 
   /**
-   * Construct a row from previously seen partial rows. This method may only be invoked
-   * when isRowCommitted returns true indicating a COMMIT_ROW chunk has been encountered.
+   * Construct a row from previously seen partial rows. This method may only be invoked when
+   * isRowCommitted returns true indicating a COMMIT_ROW chunk has been encountered.
    */
   public Row buildRow() {
     Preconditions.checkState(committed,
-        "Cannot build a Row object if we have not yet encountered a COMMIT_ROW chunk.");
-    Row.Builder currentRowBuilder = Row.newBuilder();
-    currentRowBuilder.setKey(currentRowKey);
-    for (Family.Builder builder : familyMap.values()) {
-      currentRowBuilder.addFamilies(builder.build());
+      "Cannot build a Row object if we have not yet encountered a COMMIT_ROW chunk.");
+    Row.Builder rowBuilder = Row.newBuilder().setKey(currentRowKey);
+    if (!families.isEmpty()) {
```

Comment:
I agree that we should merge families at the end, but why looping through families multiple times?
Can we do following?
Map<String, <Family.Builder> family_builder_map ...
for (Family family : families) {
  Family.Builder builder = family_builder_map.get(family.getName());
  if (builder == null) {
    builder = Family.newBuilder().setName(family.getName());
  }
  builder.addFamilies(family.getColumnsList());
}